### ISSUE TITLE
Unable to import v0.69.0 of github.com/prometheus-operator/prometheus-operator

### ISSUE THREAD

**Issue by jaronoff97** [prometheus-operator/prometheus-operator#6069]
**What happened?**

I'm trying to upgrade to v0.69.0 for these dependencies
```
	github.com/prometheus-operator/prometheus-operator v0.68.0
	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.68.0
```

but it seems that [this PR](https://github.com/prometheus-operator/prometheus-operator/pull/5956) introduced malformed filenames on mac, resulting in this error locally:

```
⫸ go mod tidy
go: downloading github.com/prometheus-operator/prometheus-operator v0.69.0
github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/watcher imports
	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1: create zip: pkg/prometheus/testdata/TSDB_config_<_v2.39.0.golden: malformed file path "pkg/prometheus/testdata/TSDB_config_<_v2.39.0.golden": invalid char '<'
pkg/prometheus/testdata/TSDB_config_>=_v2.39.0.golden: malformed file path "pkg/prometheus/testdata/TSDB_config_>=_v2.39.0.golden": invalid char '>'
github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/watcher imports
	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1: create zip: pkg/prometheus/testdata/TSDB_config_<_v2.39.0.golden: malformed file path "pkg/prometheus/testdata/TSDB_config_<_v2.39.0.golden": invalid char '<'
pkg/prometheus/testdata/TSDB_config_>=_v2.39.0.golden: malformed file path "pkg/prometheus/testdata/TSDB_config_>=_v2.39.0.golden": invalid char '>'
```

**Did you expect to see something different?**
I expected to download the new modules.

**How to reproduce it (as minimally and precisely as possible)**:
* use a mac
* have a dependency on prometheus-operator with v0.69
* `go mod tidy`

**Environment**
mac os

**Comment by jaronoff97** [comment]
A colleague tested this and this also happens on linux

**Comment by simonpasquier** [comment]
Hmm it looks like we have to rename some of the test golden files which isn't a big deal. I'll send a PR and cut a v0.69.1 (which will also use the latest Go toolchain and get rid of CVE-2023-45283 & CVE-2023-45284).

**Comment by jaronoff97** [comment]
Closed by #6070. Thank you :D
